### PR TITLE
Selected farm view

### PR DESCRIPTION
**Pull request by mlodyFrontendowiec**

### CODE REVIEW

File: src/components/FarmsList/SelectedFarm/SelectedFarm.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                <AnimatedNumber
+                  value={value}
+                  duration={300}
+                  formatValue={(value: string) => Number(value).toFixed(2)}
+                />
+                {currency}
+              </Typography>
+            </Typography>
+            <Typography className={classes.infoText}>
+              Total Staked:
```
Comment (awojciak) [inline src/components/FarmsList/SelectedFarm/SelectedFarm.tsx:126]: why is this value repeated many times? consult with someone, what should be shown there

File: src/components/FarmsList/SelectedFarm/SelectedFarm.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+              <Typography className={classes.labelText}>
+                {isXsDown
+                  ? `$ ${value * currencyPrice}`
+                  : `${value} ${currency} = $ ${value * currencyPrice}`}
+              </Typography>
+            </Box>
+          </Grid>
+        </>
+      )}
+      {activeValue === 'stake' ? (
```
Comment (awojciak) [inline src/components/FarmsList/SelectedFarm/SelectedFarm.tsx:172]: merge this ternary with the upper one

File: src/components/FarmsList/SelectedFarmList/SelectedFarmList.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+              value={element.value}
+              staked={element.staked}
+              pair={element.pair}
+              currency={element.currency}
+              currencyPrice={element.currencyPrice}
+              apy={element.apy}
+              liquidity={element.liquidity}
+              stake={element.stake}
+              unstake={element.unstake}
+              claimRewards={element.claimRewards}
```
Comment (awojciak) [inline src/components/FarmsList/SelectedFarmList/SelectedFarmList.tsx:73]: i don't think each element will have it's own claim rewards callback, probably there will be one callback from props with some argument, for example index of farm or something similar

File: src/components/FarmsList/SelectedFarmList/SelectedFarmList.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        </Typography>
+        <Typography
+          className={classes.postionsInfo}
+          style={{ display: 'flex', alignItems: 'center' }}>
+          Reward:
+          <Typography
+            component='span'
+            className={classes.value}
+            style={{ display: 'flex', alignItems: 'center' }}>
+            <img src={iconSNY} alt={'SNY token icon'} className={classes.smallIcon} /> {rewards} /
```
Comment (awojciak) [inline src/components/FarmsList/SelectedFarmList/SelectedFarmList.tsx:55]: I don't think SNY will be rewards token, maybe set this through props

File: src/components/FarmsList/SelectedFarm/SelectedFarm.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+import React, { useState } from 'react'
+import useStyle from './style'
+import AnimatedNumber from '@components/AnimatedNumber'
+import { theme } from '@static/theme'
+
+export interface ISelectedFarm {
+  value: number
+  staked: number
+  pair: string
+  currency: string
```
Comment (awojciak) [inline src/components/FarmsList/SelectedFarm/SelectedFarm.tsx:11]: maybe rename this to "rewardsToken"?

File: src/components/FarmsList/SelectedFarm/SelectedFarm.tsx [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+            <Box className={classes.boxLeft}>
+              <Typography className={classes.infoText}>
+                Total Staked:
+                <Typography display='inline' component='span' className={classes.value}>
+                  <AnimatedNumber
+                    value={value}
+                    duration={300}
+                    formatValue={(value: string) => Number(value).toFixed(2)}
+                  />
+                  <span className={classes.spacing}>{currency}</span>
```
Comment (awojciak) [inline src/components/FarmsList/SelectedFarm/SelectedFarm.tsx:79]: i think total staked value will use one of symbols in pair or both

**Review by awojciak — CHANGES_REQUESTED** [review]
also fix any visual bugs

File: src/components/FarmsList/SelectedFarmList/SelectedFarmList.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+          <div className={classes.tile}>
+            <SelectedFarm
+              value={element.value}
+              staked={element.staked}
+              pair={element.pair}
+              rewardsToken={element.rewardsToken}
+              currencyPrice={element.currencyPrice}
+              apy={element.apy}
+              liquidity={element.liquidity}
+              handleFarm={element.handleFarm}
```
Comment (awojciak) [inline src/components/FarmsList/SelectedFarmList/SelectedFarmList.tsx:73]: it's not what i meant, I wanted just to move callbacks outside elements array

Comment (mlodyFrontendowiec) [reply src/components/FarmsList/SelectedFarmList/SelectedFarmList.tsx]: Fixed, I threw a function into a component and it is called as a callback with different parameters depending on the action

File: src/components/FarmsList/SelectedFarm/SelectedFarm.tsx [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+  const handleButtonStake = (value: string) => {
+    SetActiveValue(value)
+  }
+  const isXsDown = useMediaQuery(theme.breakpoints.down('xs'))
+  return (
+    <Grid className={classes.root} container direction='column'>
+      <Grid className={classes.top}>
+        <Box className={classes.buttonContainer}>
+          <Button
+            onClick={() => handleButtonStake('stake')}
```
Comment (awojciak) [inline src/components/FarmsList/SelectedFarm/SelectedFarm.tsx:40]: there should be some action from props, not only setting something in state

File: src/components/FarmsList/SelectedFarmList/SelectedFarmList.tsx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+            <SelectedFarm
+              key={index}
+              value={element.value}
+              staked={element.staked}
+              pair={element.pair}
+              rewardsToken={element.rewardsToken}
+              currencyPrice={element.currencyPrice}
+              apy={element.apy}
+              liquidity={element.liquidity}
+              action={'stake'}
```
Comment (awojciak) [inline src/components/FarmsList/SelectedFarmList/SelectedFarmList.tsx:74]: there should be some action handlers in props, clicking on button now does nothing

**Review by awojciak — APPROVED** [review]

**Pull request merged**